### PR TITLE
Video component

### PR DESCRIPTION
**Pull request by emettely**
**Is your Pull Request request related to another [issue](https://github.com/bbc/digital-paper-edit-storybook/issues) in this repository ?**      

https://github.com/bbc/digital-paper-edit-storybook/issues/23

**Describe what the PR does**    
Refactor work and creates a video component in the storybook, using video context.

**State whether the PR is ready for review or whether it needs extra work**    
ready

**Additional context**

### CODE REVIEW

File: packages/components/PreviewCanvas/Controls.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          size="sm"
+          block
+          variant="outline-secondary"
+          onClick={ isPlaying ? handlePause : handlePlay }
+        >
+          { isPlaying ? pauseIcon : playIcon }
+        </Button>
+      </Col>
+      <Col
+        sm={ 6 } md={ 6 } ld={ 6 } xl={ 6 }
```
Comment (emettely) [inline packages/components/PreviewCanvas/Controls.js]:
```suggestion
        sm={ 6 }
```

File: packages/components/PreviewCanvas/Filler.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const Filler = (props) => {
+  return <div style={ {
+    width: `${ props.percentage }%`,
+    background: 'red',
+    height: '100%',
+    borderRadius: 'inherit'
+  } }></div>;
```
Comment (emettely) [inline packages/components/PreviewCanvas/Filler.js]:
```suggestion
  } } />;
```

File: packages/components/PreviewCanvas/ProgressBar.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import Filler from './Filler';
+// import styles from './ProgressBar.module.css';
```
Comment (emettely) [inline packages/components/PreviewCanvas/ProgressBar.js]:
```suggestion
```

File: packages/components/PreviewCanvas/ProgressBar.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+const ProgressBar = (props) => {
+
+  return (
+    // <div onClick={ props.onClick } className={ styles.progressBar }>
```
Comment (emettely) [inline packages/components/PreviewCanvas/ProgressBar.js]:
```suggestion
```

File: packages/components/PreviewCanvas/VideoContextPreview/Canvas.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable react/prop-types */
+import React, { useRef, useEffect, useState } from 'react';
+
+const Canvas = (props) => {
+  const ref = props.ref;
+
+  // const [ videoContext, setVideoContext ] = useState();
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/Canvas.js]:
```suggestion
```

File: packages/components/PreviewCanvas/VideoContextPreview/Canvas.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable react/prop-types */
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/Canvas.js]:
```suggestion
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+const VideoContextProgressBar = (props) => {
+  const ref = useRef();
+
+  const [ percentage, setPercentage ] = useState(0);
+  const [ videoContext, setVideoContext ] = useState();
+  const [ width, setWidth ] = useState(0);
+
+  const handleClick = ({ nativeEvent: { offsetX } }) => {
+    const currentTime = (offsetX / width) * videoContext.duration;
+    videoContext.currentTime = currentTime;
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
    videoContext.currentTime = (offsetX / width) * videoContext.duration;
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+const VideoContextProgressBar = (props) => {
+  const ref = useRef();
+
+  const [ percentage, setPercentage ] = useState(0);
+  const [ videoContext, setVideoContext ] = useState();
+  const [ width, setWidth ] = useState(0);
+
+  const handleClick = ({ nativeEvent: { offsetX } }) => {
+    const currentTime = (offsetX / width) * videoContext.duration;
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    videoContext.currentTime = currentTime;
+    const p = getPercentage(videoContext.currentTime, videoContext.duration);
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
    const percent = getPercentage(videoContext.currentTime, videoContext.duration);
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    setPercentage(p);
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
    setPercentage(percent);
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setWidth(ref.current.offsetWidth);
+    }
+  }, [ ref ]);
+
+  useEffect(() => {
+    const fillerAnimation = () => {
+      const p = getPercentage(videoContext.currentTime, videoContext.duration);
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
      const percent = getPercentage(videoContext.currentTime, videoContext.duration);
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+      setPercentage(p);
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.js]:
```suggestion
      setPercentage(percent);
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.test.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+import VideoContextProgressBar from './VideoContextProgressBar';
+
+describe('VideoContextProgressBar', () => {
+  const mockVideoContext = {};
+
+  it('should render correctly', () => { const component = shallow(<VideoContextProgressBar videoContext={ mockVideoContext } />);
+    expect(component).toMatchSnapshot();
+  });
+
+  it('should be possible to activate button with Spacebar', () => {
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.test.js]:
```suggestion
  it('should fully render', () => {
```

File: packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.test.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import VideoContextProgressBar from './VideoContextProgressBar';
+
+describe('VideoContextProgressBar', () => {
+  const mockVideoContext = {};
+
+  it('should render correctly', () => { const component = shallow(<VideoContextProgressBar videoContext={ mockVideoContext } />);
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/VideoContextProgressBar.test.js]:
```suggestion
  it('should render shallowly', () => { const component = shallow(<VideoContextProgressBar videoContext={ mockVideoContext } />);
```

File: packages/components/PreviewCanvas/VideoContextPreview/index.js [hunk 15]
```diff
@@ -0,0 +1,86 @@
+/* eslint-disable react/prop-types */
```
Comment (emettely) [inline packages/components/PreviewCanvas/VideoContextPreview/index.js]:
```suggestion
```

File: packages/components/PreviewCanvas/stories/index.stories.js [hunk 16]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { storiesOf } from '@storybook/react';
+import PreviewCanvas from '../index.js';
+
+const playlist = [
+  { type:'video', start:0, sourceStart: 30, duration:10, src:'https://download.ted.com/talks/MorganVague_2018X.mp4' },
+  // { type:'video', start:10, sourceStart: 40, duration:10, src:'https://download.ted.com/talks/IvanPoupyrev_2019.mp4' },
```
Comment (emettely) [inline packages/components/PreviewCanvas/stories/index.stories.js]:
```suggestion
  { type:'video', start:10, sourceStart: 40, duration:10, src:'https://download.ted.com/talks/IvanPoupyrev_2019.mp4' },
```

File: packages/components/PreviewCanvas/stories/index.stories.js [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+  // { type:'video', start:20, sourceStart: 50, duration:10, src:'https://download.ted.com/talks/KateDarling_2018S-950k.mp4' },
```
Comment (emettely) [inline packages/components/PreviewCanvas/stories/index.stories.js]:
```suggestion
  { type:'video', start:20, sourceStart: 50, duration:10, src:'https://download.ted.com/talks/KateDarling_2018S-950k.mp4' },
```

File: packages/components/PreviewCanvas/Controls.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  const handleStop = () => {
+    props.handleStop();
+    setIsPlaying(false);
+  };
+
+  return (
+    <>
+      <Col
```
Comment (allishultes) [inline packages/components/PreviewCanvas/Controls.js]:
```suggestion
      <Col sm={ 6 }>
```

**Comment by emettely** [comment]
> Looks good - not sure I've made this comment before, but noticing we're using Bootstrap classes to style more than just the grid and also inline styles. We might want to refactor CSS to a different format at some point in the future (but think we can worry about it later)

Yup good point - leaving that to a later point but agree it is something we can look at.

**Pull request merged**